### PR TITLE
feat(ext): add support for flag type and rollout definitions

### PR DESCRIPTION
**Pull request by GeorgeMac**
Fixes FLI-422

This adds support for the new flag type and rollouts support to the `import/export` format.
This is based on the eval v2 work currently in progress.

It advances the latest supported flag state format version to `1.1`.
Support for both type and rollouts is only included when version is `>=1.1`.
This is enforced on `import`.

Rollout rank is implied by order in imported document.
Rule rank can now also be omitted when using version 1.1 and it will be implied based on order.

Outstanding

- [x] Update `import/export` readonly integration tests with rollout cases.
- [x] Answer oustanding questions around renaming percentage rollout type to threshold.
- [x] Figure out if we want to support implicit rank on existing rules in `>=1.1`.

### CODE REVIEW

File: internal/ext/exporter.go [hunk 1]
```diff
@@ -6,19 +6,28 @@ import (
...
 	"gopkg.in/yaml.v2"
 )
 
 const (
 	defaultBatchSize = 25
-	version          = "1.0"
+)
+
+var (
+	latestVersion     = semver.Version{Major: 1, Minor: 1}
```
Comment (markphelps) [inline internal/ext/exporter.go:19]: nice

**Review by markphelps — APPROVED** [review]
looks great, we should also update the `cue` definition. maybe we need another issue / PR to track? is there a way to support versioning in that though? like enforcing allowed values in the document based on the `version`?

File: build/testing/integration/readonly/readonly_test.go [hunk 2]
```diff
@@ -128,8 +128,8 @@ func TestReadOnly(t *testing.T) {
 					require.NoError(t, err)
 
 					if flags.NextPageToken == "" {
-						// ensure last page contains 1 entry
-						assert.Len(t, flags.Flags, 1)
+						// ensure last page contains 2 entres (boolean and disabled)
```
Comment (yquansah) [inline build/testing/integration/readonly/readonly_test.go:131]:
```suggestion
						// ensure last page contains 2 entries (boolean and disabled)
```

File: build/testing/integration.go [hunk 3]
```diff
@@ -272,7 +272,7 @@ func importExport(ctx context.Context, _ *dagger.Client, base, flipt *dagger.Con
 
 		diff := cmp.Diff(expected, generated)
 		if diff != "" {
-			fmt.Println("Unexpected difference in exported output:")
+			fmt.Printf("Unexpected difference in %q exported output:\n", conf.name)
```
Comment (yquansah) [inline build/testing/integration.go:275]:
```suggestion
			fmt.Printf("Unexpected difference in %q exported output: \n", conf.name)
```

File: internal/ext/importer.go [hunk 4]
```diff
@@ -246,6 +282,53 @@ func (i *Importer) Import(ctx context.Context, r io.Reader) error {
...
+			}
+
+			for idx, r := range f.Rollouts {
+				if r.Segment != nil && r.Threshold != nil {
+					return fmt.Errorf(`rollout "%s/%s/%d" cannot have both segment and percentage rule`,
+						namespace,
+						f.Key,
+						idx,
+					)
+				}
```
Comment (yquansah) [inline internal/ext/importer.go:302]: I wonder if we can also statically detect this.. note for the `flipt validate`!

**Review by yquansah — APPROVED** [review]

File: internal/cue/fixtures/valid.yaml [hunk 5]
```diff
@@ -11,15 +11,26 @@ flags:
     name: flipt
   rules:
   - segment: internal-users
-    rank: 1
```
Comment (markphelps) [inline internal/cue/fixtures/valid.yaml:14]: just FYI, `fixtures` has been renamed to `testdata` on main so that it works with fuzz testing

**Review by markphelps — APPROVED** [review]
nice x2

**Review by yquansah — APPROVED** [review]

**Comment by GeorgeMac** [comment]
To recap on the failing tests as I am merging this into `eval-v2`:
1. DBs other than SQLite fail as we haven't prioritized implementing those yet.
2. Migration tests as they depend upon readonly tests the current branch. We need to leverage the readonly tests from the current release instead (as these should be forward compatible from last release, where new tests we add to readonly cover cases not supported by older Versions of Flipt).
3. The FS integration tests as we haven't started work on the storage backend for them (Yoofi is starting that soon).

**Pull request merged**